src/_pytest/logging.py

```diff
@@ -72,24 +72,62 @@ class PercentStyleMultiline(logging.PercentStyle):
     formats the message as if each line were logged separately.
     """
 
+    def __init__(self, fmt, auto_indent):
+        super().__init__(fmt)
+        self._auto_indent = auto_indent
+
     @staticmethod
     def _update_message(record_dict, message):
         tmp = record_dict.copy()
         tmp["message"] = message
         return tmp
 
+    def _get_auto_indent(self, record):
+        if hasattr(record, "auto_indent"):
+            # passed in from the "extra={}" kwarg on the call to log.log()
+            auto_indent = record.auto_indent
+        elif self._auto_indent is not None:
+            auto_indent = self._auto_indent
+        else:
+            auto_indent = -1
+
+        if type(auto_indent) is str:
+            if auto_indent.lower() == "true" or auto_indent.lower() == "on":
+                auto_indent = -1
+            elif auto_indent.lower() == "false" or auto_indent.lower() == "off":
+                auto_indent = 0
+            else:
+                try:
+                    auto_indent = int(auto_indent)
+                except ValueError:
+                    auto_indent = -1
+        elif type(auto_indent) is bool:
+            if auto_indent:
+                auto_indent = -1
+            else:
+                auto_indent = 0
+
+        if type(auto_indent) is not int:
+            auto_indent = -1
+        return auto_indent
+
     def format(self, record):
         if "\n" in record.message:
-            lines = record.message.splitlines()
-            formatted = self._fmt % self._update_message(record.__dict__, lines[0])
-            # TODO optimize this by introducing an option that tells the
-            # logging framework that the indentation doesn't
-            # change. This allows to compute the indentation only once.
-            indentation = _remove_ansi_escape_sequences(formatted).find(lines[0])
-            lines[0] = formatted
-            return ("\n" + " " * indentation).join(lines)
-        else:
-            return self._fmt % record.__dict__
+            auto_indent = self._get_auto_indent(record)
```

Comment:
Just to be perfectly clear, per my suggestion above this becomes:

```python
auto_indent = get_auto_indent(record, self._auto_indent)
```

Reply:
Understood